### ISSUE TITLE
Clarify in which contexts `$_`  and its alias, `$PSItem`, are meaningfully defined

### ISSUE THREAD

**Issue by mklement0** [MicrosoftDocs/PowerShell-Docs#9521]
### Prerequisites

- [X] **Existing Issue:** Search the existing issues for this repository. If there is an issue that fits your needs do not file a new one. Subscribe, react, or comment on that issue instead.
- [X] **Descriptive Title:** Write the title for this issue as a short synopsis. If possible, provide context. For example, "Typo in `Get-Foo` cmdlet" instead of "Typo."
- [X] **Verify Version:** If there is a mismatch between documentation and the behavior on your system, ensure that the version you are using is the same as the documentation. Check this box if they match or the issue you are reporting is not version specific.

### Links

- https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_Automatic_Variables?view=powershell-7.3#_

### Summary

The linked descriptions of `$_` (as well as that of `$PSItem`) do not explain in which contexts these variables are _meaningfully_ defined, leading users to attempt to use them in inappropriate contexts - see [this StackOverflow question](https://stackoverflow.com/q/74680073/45375) for an example.

### Details

In short:

* `$_` / `$PSItem` are only ever meaningful _inside script blocks_ ...
* ... used in _specific contexts_.

The specific contexts are worth enumerating.

The bottom section of [this StackOverflow answer](https://stackoverflow.com/a/55667981/45375) tries to do just that.
